src/ebonite/core/objects/wrapper.py

```diff
@@ -383,6 +383,11 @@ def _safe_analyze(self, obj):
         :param obj: object to check
         :return: :class:`ModelIO` instance or None
         """
+
+        # avoid calling heavy ModelAnalyzer machinery for built-in objects
+        if isinstance(obj, (bool, int, float, complex, str, list, tuple, set, dict, type, type(None))):
```

Comment:
maybe there will be even more efficiency if we use whitelist instead of blacklist? I mean smth like add class attribute to hook type in which you will list every type that "might" be right?

Reply:
and here you can check that obj is an instance of a type from a union of those lists

Reply:
I think that whitelist: (1) poses additional mental overhead to implementers of new model wrappers, it's not so easy to understand which type "is right"; (2) brings us back to work with `Hook`s  via at least iteration over them which we try to avoid here; (3, most important) spreads this optimization through a code base instead of explicitly encapsulating and documenting it here.

Reply:
Sorry but I don't get your remark about "union of lists". What do you mean by that?

Reply:
(1) is a valid point, but it is not yet clear if it outweights the benefits of whitelist. (2) we can do it once, or better just maintain a list of types and update it on new hook registration (3) in my mind it still happens here, we just use a dynamic list instead of hardcoded one.
Here are my points: 1) we can have some strange case when used decides that dict is a valid model type (for example, very big mapping table), writes wrapper and io that does some heavy compressing. and we will skip his code. that is the point against hardcoded blacklist 2) blacklist is simply not very efficient, because there is A LOT of other types, that can never be a model. for example, every single class from standard library. or every single class from every single other library, except a couple of classes from ML libraries. this sounds like a job for whitelist)) 3) most of ModelHooks already utilizes TypeHook mixin (or dont utilize, but should), which is exactly what i am proposing. there are 3 that do not, it's TFTensor (which by the way can fail with whitelist code), Sklearn (it just checks if object is from sklean module, which can be wrong, if you think about it) and Callable, which cannot use instance check. but it is builtin, so we can make an exception and check it separately.
generally speaking, coupling modelhooks with a list of types makes sense, because i cant think of an example, where hook should work for arbitrary type

Reply:
fixed